src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -51,42 +49,44 @@
 
     public static String TMP_EXT = ".tmp";
 
-    private static final Splitter filenameSplitter = Splitter.on('-');
+    public static final char FILENAME_SEPARATOR = '-';
+
+    private static final Splitter filenameSplitter = Splitter.on(FILENAME_SEPARATOR);
 
     /** canonicalized path to the directory where SSTable resides */
     public final File directory;
     /** version has the following format: <code>[a-z]+</code> */
     public final Version version;
     public final String ksname;
     public final String cfname;
-    public final int generation;
+    public final SSTableId generation;
```

Comment:
We could call this attribute `sstableId`, or even `id`, now that `generation` is an attribute of one of the implementations of the more generic `SSTableId` interface. The same applies to the method parameters named `generation`, local variables and mentions in comments.

Reply:
hmm.... this will make this commit quite large comparing to what it does; if we really need it, I'd not do that in this PR

Reply:
I don't think it adds that much complexity, and it seems natural to change the names of attributes and parameters in the same commit where we change their type. I gave it a quick try in [this commit](https://github.com/adelapena/cassandra/commit/3afbb0f76f8f0e2d95c8dbe15717f6412d3a2d62), just to see how it would look like.

Reply:
@blerer what do you think about this? Are you ok with renaming `generation` to `id`/`sstableid` in a separate ticket? I guess I can live with it.